src/handlers/extensions/extensionInitialised.ts

```diff
@@ -21,9 +31,15 @@ export default async (event: ContractEvent): Promise<void> => {
     },
   });
 
+  let removeMotionListeners: RemoveListener[] = [];
   /* Listen for motions once Voting Reputation is enabled. */
   if (getExtensionHash(Extension.VotingReputation) === extensionHash) {
-    await motionSpecificEventsListener(colonyAddress);
+    removeMotionListeners = await motionSpecificEventsListener(colonyAddress);
+
+    /* Store remove listener functions in context, to be called when extension is uninstalled */
+    context.removeListeners[colonyAddress] = {
```

Comment:
There's a few issues I can see with this approach:
1. Introducing extra abstraction in a form of context that is being passed around as argument, which I think is unnecessary. You can simply export/import whatever variable that holds the removers (or anything other piece of data you want to share between files). You are not restricted by React state and how it impacts the rendering.
```js
  // a.ts
  export const removers = {};
  
  // b.ts 
  import { removers } from './a';

  removers.add();
  removers.remove();
```
Obviously you can add some methods wrapping the logic.

2. I think too many pieces are currently aware of the context and event removers. `eventProcessor` doesn't need to know anything about them, and they could be isolated in their own file/module
3. It requires the listener to return a specific object, which has to be then handled properly by the extension initialised handler and extension uninstalled. Again, too many things know too much about the implementation details of this solution.

It's difficult to propose something concrete without looking into it further, but the general idea I would try to follow would be:
- Figure out a way of uniquely identifying the listeners. It seems like we want to remove the listeners for all events for a particular contract address (in this case the voting reputation client address), so maybe that is enough?
- Have an object mapping the unique identifiers to the remover functions, and some methods to add and trigger those 
- Add the remover function to the object at the "source", ie. the `eventListenerGenerator` method
- Call the trigger method passing it the unique identifier whenever you need to remove a listener

The advantages of that would be that no other function (e.g. `motionSpecificEventsListener`, `addMotionEventListener`) needs to know about the concept of removing listeners and you can encapsulate all the complexity in one place

It might not work without further adjustments but I'll be happy to jump on a call (and it'd make sense to get Raul on board too) and figure something out together